scripts/restrict-via-common-ancestor.py

```diff
@@ -0,0 +1,83 @@
+"""
+Returns a monophyletic clade from the provided tree which encompasses all the (terminal)
+strains provided via a list or an Auspice dataset.
+"""
+
+import argparse
+import json
+from Bio import Phylo
+from sys import exit
+from typing import cast, Set, Dict, Any
+from augur.io import read_strains
+
+
+def strains_in_auspice_json(fname: str) -> Set[str]:
+    with open(fname) as fh:
+        j = json.load(fh)
+    assert isinstance(j['tree'], dict), "Only supports datasets with a single dataset"
+    strains = set()
+    nodes = [j['tree']]
+    while len(nodes):
+        node = nodes.pop(0)
+        children = node.get("children", [])
+        nodes.extend(children)
+        if not children:
+            strains.add(node['name'])
+    return strains
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--tree', type=str, metavar="NEWICK", required=True, help="Input tree")
+    parser.add_argument('--strains', type=str, metavar="TXT|JSON", required = True,
+        help="A list of strains or Auspice JSON whose strains are used to define the common ancestor")
+    parser.add_argument('--output-tree', type=str, metavar="NEWICK", required = True, help="output tree")
+    parser.add_argument('--output-metadata', type=str, metavar="JSON", help="output node-data JSON")
+    args = parser.parse_args()
+
+
+    tree = Phylo.read(args.tree, "newick")
+    if args.strains.lower().endswith(".txt"):
+        all_defining_strains = cast(Set[str], read_strains(args.strains))
+    elif args.strains.lower().endswith(".json"):        
+        all_defining_strains = strains_in_auspice_json(args.strains)
+    else:
+        print("Unknown format of '--strains' - must be .txt or .json")
```

Comment:
_non-blocking nit_

```suggestion
        print_err("Unknown format of '--strains' - must be .txt or .json")
```

and add `print_err` to the `augur.io` import at the top.

Reply:
👍 Changed these to now print to stderr, which may be useful for logging purposes in the future. Maybe?